edi-tools/modules/codegen/rest_connector_gen.bal

```diff
@@ -0,0 +1,36 @@
+
+function generateRESTConnector(string libName) returns string {
+    string restConCode = string `
+import ballerina/http;
+import ballerina/edi;
+
+configurable int restConnectorPort = 9090;
+
+service /${libName}EDI on new http:Listener(restConnectorPort) {
+
+    isolated resource function post reader/[string ediType](@http:Payload string ediData) returns json|error {
+        EDI_NAME|error ediTypeName = ediType.ensureType();
+        if ediTypeName is error {
+            return error edi:Error("Unsupported EDI type: " + ediType + ". " + ediTypeName.message());
```

Comment:
@MaryamZi is there a way to write this as `return error edi:Error(string `Unsupported EDI type: ${ediType}.${ediTypeName.message()});` Effectivly this means string template within another string template.

Reply:
Yes, as I can remember, it didn't work. May be I was not using the correct escape character.

Reply:
Will have to do something like
```suggestion
            return error edi:Error(string ${"`"}Unsupported EDI type: ${"${ediType}"}. ${"${ediTypeName.message()}`"});
```

Generated code will be

```ballerina
return error edi:Error(string `Unsupported EDI type: ${ediType}. ${ediTypeName.message()}`);
```